src/sentry/api/endpoints/group_reprocessing.py

```diff
@@ -0,0 +1,39 @@
+from __future__ import absolute_import
+
+
+from sentry import features
+from sentry.api.base import DocSection
+from sentry.api.bases import GroupEndpoint
+from sentry.models import Group
+from sentry.utils.apidocs import scenario, attach_scenarios
+from sentry.tasks.reprocessing2 import reprocess_group
+
+
+@scenario("ReprocessGroup")
+def reprocess_group_scenario(runner):
+    group = Group.objects.filter(project=runner.default_project).first()
+    runner.request(method="PUT", path="/issues/%s/reprocessing/" % group.id)
+
+
+class GroupReprocessingEndpoint(GroupEndpoint):
+    doc_section = DocSection.EVENTS
+
+    @attach_scenarios([reprocess_group_scenario])
+    def put(self, request, group):
```

Comment:
Should this be `POST` rather? At the moment, reprocessing is not idempotent (it does create new events every time), so this resembles `POST` semantics.

Reply:
I've seen PUT used to create new entities, so I thought "let's make a reprocessing task an entity". TIL PUT is supposed to be idempotent. Will fix.

Reply:
Yeah. The idea is that you use `PUT` on resources that you can name and identify, mostly overwriting their (entire) data. For example, writing a file to a location. Some frameworks like Rails also use `PATCH` to do partial updates.

`POST`, on the other hand, is supposed to add to a collection, and you can add the same data multiple times resulting in multiple entries. PUT should not have that behavior.